File: 007_boundary/solution/boundary_length.m
function L = boundary_length(V,F)
%BOUNDARY_LENGTH Compute the length of a mesh's boundary
%
% Input:
%  V,F  input mesh
% Output:
%  L  the length of the boundary

%Compute the indices of all boundary edges.
O = outline(F);

%Compute a matrix whose rows are equal to edgeEnd-edgeStart for each
%boundary edge.
BE = V(O(:,2),:) - V(O(:,1),:);

%Compute the total length of all boundary edges.
L = sum(normrow(BE));

end

File: 007_boundary/solution/boundary_length.py
import gpytoolbox as gpy, numpy as np

def boundary_length(V,F):
    """Compute the length of the input mesh V,F's boundary.
    """

    # Compute the boundary edges
    bdry_edges = gpy.boundary_edges(F)

    # Compute a matrix whose rows are equal to edgeEnd-edgeStart for each
    BE = V[bdry_edges[:,1],:] - V[bdry_edges[:,0],:]

    # Compute the total length of all boundary edges
    # HINT: look up the documentation for the NumPy function np.linalg.norm, and
    #   study the `axis` parameter in particular.
    L = np.sum(np.linalg.norm(BE, axis=-1))

    return L
